### PR TITLE
Introduce item B008

### PR DESCRIPTION
**Pull request by emranbm**
Based on our discussion at Sahab Quality Committee, I've prepared a recommendation for a new maturity item at level 4:
> CI/CD agents installation and config should be in version control.

### CODE REVIEW

File: catalog/build_ci/B008.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### B008
+
+|**Code**           | **B008** |
+|**Title**          | CI/CD runner servers should be set up automatically.|
```
Comment (hamidrezash2000) [inline catalog/build_ci/B008.md:4]:
```suggestion
|**Title**          | CI/CD agents installation and config should be in version control.|
```
I think it's title and description should be more abstract.
They might use server or whatever.
They might not setup the agent automatically, but the instructions or dependencies are specified clearly.
Consider a scenario that a team works with Github actions or they got their agents as service from another team

File: LEVELS.md [hunk 2]
```diff
@@ -50,6 +50,7 @@ Here are the maturity levels. This is what we use as maturity levels and you may
 |**[T014]({{ site.baseurl }}{% link catalog/test/T014.md%})**| Requirement - Acceptance Test Mapping > 20%. |
 |**[B006]({{ site.baseurl }}{% link catalog/build_ci/B006.md%})**| Actively Use Available Analysis Tools (80% sonar-way). |
 |**[B007]({{ site.baseurl }}{% link catalog/build_ci/B007.md%})**| Build Environment Can Be Automatically Created From Version Control. |
+|**[B008]({{ site.baseurl }}{% link catalog/delivery/B008.md%})**| CI/CD runner servers should be set up automatically. |
```
Comment (hamidrezash2000) [inline LEVELS.md:53]: https://github.com/qualiticcommunity/qualiticcommunity.github.io/pull/15#discussion_r763623713

File: catalog/build_ci/B008.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+### B008
+
+|**Code**           | **B008** |
+|**Title**          | CI/CD agents installation and config should be in version control.|
+|**Description**    | There should exist scripts to setup the CI runner(s) from scratch.|
```
Comment (hamidrezash2000) [inline catalog/build_ci/B008.md:5]: https://github.com/qualiticcommunity/qualiticcommunity.github.io/pull/15#discussion_r763623713

**Review by hamidrezash2000 — CHANGES_REQUESTED** [review]

**Review by mojyou — DISMISSED** [review]

File: catalog/build_ci/B008.md (hunk 3 above)
Comment (emranbm) [reply catalog/build_ci/B008.md]: Please see if that's OK now.

**Review by mojyou — APPROVED** [review]

**Review by hamidrezash2000 — APPROVED** [review]

**Review by hamidrezash2000 — APPROVED** [review]

**Review by hamidrezash2000 — APPROVED** [review]

**Review by littleali — DISMISSED** [review]

**Review by hamidrezash2000 — APPROVED** [review]

File: catalog/build_ci/B008.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+### B008
+
+|**Code**           | **B008** |
+|**Title**          | CI/CD agents installation and config should be in version control.|
+|**Description**    | Instuctions on how to setup the CI/CD agent(s) from scratch should be specified. i.e. there is no tacit knowledge about how to setup/install/configure an agent for the project.|
+|**Best Practices** | An interactive script at the root of repo that asks for a bare server info (e.g. ip/user/password) and then sets up runners on the specified targets automatically.|
```
Comment (bozorgzadeh) [inline catalog/build_ci/B008.md:6]: Maybe it is not the only practice or the best one.
Actually we have done another solution. We run pipelines in several dockers and put the Dockerfile in code base and also put the produced docker images to our artifact repository. So we do not need any manual configuration of the agent machine.

I suggest changing the title to:

There is no manual configuration of the CI/CD agent machines. Provisioning the required environment of the pipeline agents is derived from code base.

Comment (emranbm) [reply catalog/build_ci/B008.md:6]: Agree about the title.  
But about your above-mentioned approach, I think more parameters exist to be considered when setting up a CI agent **from scratch** (where we're not using any third-party pre-configured CI ecosystem). For example, how do you manage the docker daemon config itself? (e.g. proxy address, custom certificates, caches, max concurrent jobs, etc.)  
I've mentioned some flaws of just relying on docker images in such a case [here](https://vrgl.ir/Gd5zq).  
So I recommend describing in the item that there should be a non-manual script to set up the CI agent(s) from scratch **if not using CI as a (third-party) service**.

**Review by bozorgzadeh — DISMISSED** [review]

**Review by MSDehghan — DISMISSED** [review]

**Comment by emranbm** [comment]
Just rebased and resolved conflicts.

**Review by mojyou — APPROVED** [review]
Thanks Emran.

**Pull request merged**